### ISSUE TITLE
readdir mode should be inside the options

### ISSUE THREAD

**Issue by perry-mitchell** [perry-mitchell/webdav-fs#40]
Deprecate mode at the end of the function and place it inside the options object.

**Comment by artemkloko** [comment]
:+1:  This would allow to promisify the `readdir`
```js
const { promisify } = require( 'util' );
const fs = require( 'webdav-fs' )( 'xxx', 'xxx', 'xxx' );
var readdir = promisify( fs.readdir );
```

**Comment by perry-mitchell** [comment]
Available in 2.0.0
